client/PRUserial485/PRUserial485.py

```diff
@@ -161,101 +135,157 @@ def socket_communicate(conn_port, data_queue):
+    """."""
+    # create threads
+    thread_cmd_general = \
+        Thread(target=_socket_communicate,
+               args=(SERVER_PORT_GENERAL, QUEUE_GENERAL))
+    thread_cmd_general.setDaemon(True)
+    thread_cmd_rw = \
+        Thread(target=_socket_communicate,
+               args=(SERVER_PORT_RW, QUEUE_RW))
+    thread_cmd_rw.setDaemon(True)
+
+    # start threads
+    thread_cmd_general.start()
+    thread_cmd_rw.start()
+
+
+def PRUserial485_read():
+    """Recebe dados através da interface serial."""
+    # Payload: none
+    payload = _c.COMMAND_PRUserial485_read
+    command, payload_recv = _send_communication_data(payload)
+    data = [chr(i) for i in payload_recv]
+    if command == ord(_c.COMMAND_PRUserial485_read):
+        return data
     else:
         return None
 
 
-def PRUserial485_write(data=[], timeout=0):
+def PRUserial485_write(data=None, timeout=0):
     """Envia dados através da interface serial."""
     # Payload: TIMEOUT (4 bytes) + DATA (len(DATA) bytes)
+    if data is None:
+        data = []
     payload = _c.COMMAND_PRUserial485_write + struct.pack(">f", timeout)
     payload += bytearray([ord(i) for i in data])
-    command, payload_recv = send_communication_data(payload)
-    if command == ord(_c.COMMAND_PRUserial485_write) and len(payload_recv) == 1:
-        return(ord(payload_recv))
+    command, payload_recv = _send_communication_data(payload)
+    if command == ord(_c.COMMAND_PRUserial485_write) and \
+       len(payload_recv) == 1:
+        return ord(payload_recv)
     else:
         return None
 
 
-def PRUserial485_read():
-    """Recebe dados através da interface serial."""
+def PRUserial485_version():
+    """Return version."""
     # Payload: none
-    payload = _c.COMMAND_PRUserial485_read
-    command, payload_recv = send_communication_data(payload)
-    data = [chr(i) for i in payload_recv]
-    if command == ord(_c.COMMAND_PRUserial485_read):
-        return data
+    payload = _c.COMMAND_PRUserial485_version
+    command, payload_recv = _send_communication_data(payload)
+    if command == ord(_c.COMMAND_PRUserial485_version) and payload_recv:
+        return payload_recv.decode()
+    else:
+        return None
+
+
+def PRUserial485_server_version():
+    """Return version."""
+    # Payload: none
+    payload = _c.COMMAND_PRUserial485_server_eth_version
+    command, payload_recv = _send_communication_data(payload)
+    if command == ord(_c.COMMAND_PRUserial485_server_eth_version) and \
+       payload_recv:
+        return payload_recv.decode()
     else:
         return None
 
 
+def PRUserial485_address():
+    """Retorna endereco fisico da placa."""
+    # Payload: none
+    payload = _c.COMMAND_PRUserial485_address
+    command, payload_recv = _send_communication_data(payload)
+    if command == ord(_c.COMMAND_PRUserial485_address) and \
+       len(payload_recv) == 1:
+        return ord(payload_recv)
+    else:
+        return None
+
+
+# --- Deprecated functions ---
```

Comment:
These functions **are not deprecated** at all! They are implemented on server (BBB) side and working as expected, so they can be added to the library.

The difference here is that IOCs are not using them anymore.

Reply:
@pnallin ,

I did not know they were used by another client other than power supply IOCs. That's why I thought they could be discarded. if not, the library could be simplified by discarding them and thus more easily optimized. This was my rational. 

Are they going to be used by another client? If they are, I will just delete de comment line.

Reply:
@pcnallin,

In the lattest commit, that transformed the client API from module functions to class methods, I included these functions.